login_server/player/settings.py

```diff
@@ -49,7 +49,10 @@ def from_dict(cls, d):
         return cls(d.get('rank_xp', 0), last_first_win_time)
 
     def to_dict(self):
-        return {'rank_xp': self.rank_xp, 'last_first_win_time': self.last_first_win_time.isoformat()}
+        return {
+            'rank_xp': self.rank_xp,
+            'last_first_win_time': self.last_first_win_time.strftime('%Y-%m-%dT%H:%M:%S.%fZ')
```

Comment:
Just a thought. Will there be issues with the login server running in a different timezone than the game server? Maybe inconsistencies or the new day being chosen incorrectly?

Reply:
This shouldn't be an issue, because the dates for first win of the day are only generated / compared on the login server.

If we did get timezone weirdness (e.g. idk the login server changing timezones or w/e) the worst that would happen would be that the first win of the day would be given out twice in less than or more than 24 hours depending on the change.

Reply:
Doesn't that mean that NA is most likely to have 2 first wins of the day, given that their prime time is around EU's midnight? And do you care about this? I'm fine with this being a first implementation that we improve upon later.

Reply:
Still shouldn't happen because we only evaluate the time on the login server. I.e. the recorded time for the player's last win will be in EU time.

I guess an NA player might be more likely to get a win right before EU midnight and again right after, but the next day they couldn't do it again. They can still only get one first win per day.

Fwiw that was how it worked on Hirez servers as well - the cutover was about 7pm AU time which was occasionally fortuitous.

Reply:
Oh really. Nice =)